### PR TITLE
auth: include boost/regex_fwd.hpp in header 

### PR DESCRIPTION
**Pull request by tchaikov**
since we only need the full definition of boost::regex in the .cc
file, where we

- define the constructor and destructor
- and actually use the regex.

there is no need to include boost/regex.hpp in the header, in order
to keep the preprocessed header smaller. let's use a header only
contains forward declarations in header, and include the full
definition in the .cc file.

---

it's a cleanup, hence no need to backport.

### CODE REVIEW

**Comment by tchaikov** [comment]
this change depends on #20942. i will rebase this one once #20942 lands.

**Review by dawmd — APPROVED** [review]
LGTM. The vector becomes an incomplete type in the header, but we seem to rely on https://eel.is/c++draft/vector#overview-4

**Comment by tchaikov** [comment]
@scylladb/scylla-maint hello maintainers, could you help queue this change?

File: auth/certificate_authenticator.hh
Comment (nyh) [inline auth/certificate_authenticator.hh:13]: This is sad, reminds me of the ugly "noqa" lines in dtest (578 such lines in dtest, on my last count) 
Isn't this a sign we should stop using the buggy "include-cleaner"? Does it bring us any benefit?
Even the comment's name, "IWYU" ("include what you use") is a wrong - since the problem being detected here wasn't that we weren't including what we were using, but the opposite problem - that (supposedly) we included something we didn't need.

@avikivity what is your opinion on this include-cleaner thing? Is it something you wanted or like?

**Review by nyh — APPROVED** [review]
Looks good, I'm not happy about the ugly comment in the second patch (see my comments inline) but otherwise the patch is good.

Isn't it ironic that one of the things that your patch does is to remove an unused include of `<regex>`. The include-cleaner never complained about this unused header, but now complains a header which is _not_ unused but it wrongly thinks it is? It sounds like a pretty unhelpful tool :-(

**Comment by tchaikov** [comment]
> Isn't it ironic that one of the things that your patch does is to remove an unused include of `<regex>`. The include-cleaner never complained about this unused header, 

it does. see https://github.com/scylladb/scylladb/pull/20986/files . please scroll down to the bottom of the page.

> but now complains a header which is _not_ unused but it wrongly thinks it is? It sounds like a pretty unhelpful tool :-(

it complains both of them. so at least it is consistent. seems like some other headers (indirectly) include `boost/regex.hpp` for `auth/certificate_authenticator.hh`.

File: auth/certificate_authenticator.hh
Comment (avikivity) [reply auth/certificate_authenticator.hh:13]: I'm not a fan of it. But since most of the contributors aren't concerned with dependency reduction, maybe it helps a little.

**Comment by tchaikov** [comment]
merged into master. so i am closing this PR.

**Pull request closed without merging**
